server.js

```diff
@@ -1,23 +1,95 @@
 const express = require("express");
 const cors = require("cors");
-
+const { request, response } = require("express");
 const app = express();
 
+
+const bodyParser = require("body-parser")
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}))
+
 app.use(cors());
+app.use(express.json());
+
+app.use((request, response, next) => {
+  request.body.timeSent = new Date();
+  next();
+});
 
 const welcomeMessage = {
   id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
+  from: "Neill",
+  text:"hi hi"
 };
-
 //This array is our "data store".
-//We will start with one message in the array.
+//oneWe will start with  message in the array.
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+
+app.get("/messages",(request,response)=>{
+  console.log(request.body.timeSent);
+  response.status(200).json({messages})
+})
+app.get("/messages/search", (request, response) => {
+  const searchText = request.query.text;
+  let filteredMessages = messages;
+  if (searchText) {
+    filteredMessages = messages.filter((message) =>
+      message.text.includes(searchText)
+    );
+  }
+  response.json({ filteredMessages });
+});
+app.get("messages/latest", (request, response) => {
+  const searchText = request.query.text;
+  let recentMessages = messages.slice(-10);
+  if (searchText) {
+    recentMessages = recentMessages.filter((message) =>
+      message.text.includes(searchText)
+    );
+  }
+  response.json({ recentMessages });
+});
+app.get("/messages/:id",(request,response)=>{
+  const id= Number(request.params.id);
+  const find= messages.find(message=>message.id===id);
+  response.status(200).json({find})
+})
+
+
+app.post("/messages",(request,response)=>{
+   if(request.body.from==="" || request.body.text==="" ){
+    response.status(400).json({message : "fill all fields"});
+    return;
+  }
+      const newMessage= {
+    id :messages.length,
+    from : request.body.from, 
+    text : request.body.text
+  };
+  messages.push(newMessage);
+  response.status(201).json({newMessage});
+})
+
+app.delete("/messages/:id",(request,response)=>{
+  let requestId = Number(request.params.id);
+  let result = messages.filter((item) => item.id !== requestId);
+ 
+  if (requestId < 0)
+    return response.status(404).json({ msg: "message not found" });
+  response.json({result})
+
```

Comment:
1. plz don't send back the whole array/collection, plz see the possible options here: https://stackoverflow.com/questions/25970523/restful-what-should-a-delete-response-body-contain
2. plz think about whether you have really deleted anything from the messages

Reply:
If there is a collection of data then after I delete a record we should show the list without that record. am I right?